File: rust/client/src/components/shared/Waiter/Waiter.js
import React, {memo} from 'react';

import Icon from '../Icon';
import classes from './Waiter.css';

function Waiter() {
  return (
    <div className={classes.root}>
      <div className={classes.back} />
      <Icon icon="spinner" width={30} className={classes.spinner} />
    </div>
  );
}

export default memo(Waiter);

File: rust-yew/client/src/components/shared/waiter/waiter.rs
use crate::components::shared::{Icon, IconType};
use yew::function_component;
use yew::prelude::*;

#[function_component(Waiter)]
pub fn waiter() -> Html {
    log::trace!("render Waiter");
    let css = css_mod::get!("waiter.css");

    html! {
        <div class={css["root"]}>
            <div class={css["back"]} />
            <Icon icon={IconType::Spinner} class={css["spinner"]} width=40 />
        </div>
    }
}
